static/posts/pattern-matching/post.md

```diff
@@ -0,0 +1,952 @@
+---
+title: Pattern Matching
+date: 3/26/2024
+tags:
+  - programming
+  - typescript
+  - c
+  - haskell
+  - rust
+description: |
+  An overview of Pattern Matching, with demos of why it's needed, and how you
+  can use it to write cleaner and safer code
+cover:
+  alt: An AI-generated picture of a shape sorter toy
+  caption: |
+    Note: Although this image is AI generated, the rest of this post is not.
```

Comment:
Suggestion - change "the rest of this post is not" to "the rest of this post is written by a human."